src/Common/filesystemHelpers.cpp

```diff
@@ -280,14 +280,27 @@ time_t getModificationTime(const std::string & path)
     struct stat st;
     if (stat(path.c_str(), &st) == 0)
         return st.st_mtime;
-    DB::throwFromErrnoWithPath("Cannot check modification time for file: " + path, path, DB::ErrorCodes::PATH_ACCESS_DENIED);
+    DB::throwFromErrnoWithPath("Cannot modification change time for file: " + path, path, DB::ErrorCodes::PATH_ACCESS_DENIED);
+}
+
+time_t getChangeTime(const std::string & path)
+{
+    struct stat st;
+    if (stat(path.c_str(), &st) == 0)
+        return st.st_ctime;
+    DB::throwFromErrnoWithPath("Cannot check change time for file: " + path, path, DB::ErrorCodes::PATH_ACCESS_DENIED);
 }
 
 Poco::Timestamp getModificationTimestamp(const std::string & path)
 {
     return Poco::Timestamp::fromEpochTime(getModificationTime(path));
 }
 
+Poco::Timestamp getChangeTimestamp(const std::string & path)
```

Comment:
Can we avoid using `Poco::Timestamp`? It's bug-prone and doesn't have real advantages